pkg/build/api/v1beta1/types.go

```diff
@@ -207,6 +207,11 @@ type CustomBuildStrategy struct {
 	// the docker image should be pulled
 	From *kapi.ObjectReference `json:"from,omitempty"`
 
+	// PullSecretName is the name of a Secret that would be used for setting up
+	// the authentication for pulling the Docker images from the private Docker
+	// registries
+	PullSecretName string `json:pullSecretName,omitempty`
```

Comment:
the conversion will fail otherwise, @smarterclayton what is the good practice here?